web/handlers/http.go

```diff
@@ -255,16 +247,23 @@ func New(
 
 	mainMux.Handle("/", m)
 
-	// apply middleware
-	var finalHandler http.Handler = mainMux
-	finalHandler = logging.InjectHandler(logger)(finalHandler)
-	finalHandler = CSRF(finalHandler)
+	// apply HTTP middleware
+	middlewares := []func(http.Handler) http.Handler{
+		logging.InjectHandler(logger),
+		user.ContextInjecter(fs, a),
+		CSRF,
+	}
+
+	if !web.Production {
+		middlewares = append(middlewares, r.GetReloader())
+	}
 
-	if web.Production {
-		return finalHandler, nil
+	var finalHandler http.Handler = mainMux
```

Comment:
what does `finalHandler` do, just plain ol serves the http & receives the requests?

Reply:
yup. It's the handler with all the middlewares applied.